### PR TITLE
new module win_file_version

### PR DESCRIPTION
**Pull request by SamLiu79**

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @SamLiu79 for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes, which can be contributed by anyone other than the author of the new module:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

More information about the review process can be seen here: 
https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md

Thanks again for submitting your Ansible module!

File: windows/win_file_version.ps1 [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    If ($file_build_part -eq $null){
+        $file_build_part = ''
+    }
+    $file_private_part = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($path).FilePrivatePart
+    If ($file_private_part -eq $null){
+        $file_private_part = ''
+    }
+}
+Catch{
+}
```
Comment (nitzmahone) [inline windows/win_file_version.ps1:73]: Definitely shouldn't be swallowing the exception here... It's probably best to just get rid of the Try/Catch entirely, unless you're going to do a lot of extra work to communicate the error back in a useful way with Fail-Json.

**Comment by nitzmahone** [comment]
@SamLiu79 Looks pretty good, other than the exception swallowing. Fix that, and we'll merge...

**Comment by gregdek** [comment]
@SamLiu79 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by SamLiu79** [comment]
@nitzmahone @gregdek , please review. thanks.

File: windows/win_file_version.ps1 [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    $file_minor_part = $_version_fields.FileMinorPart
+    If ($file_minor_part -eq $null){
+        $file_minor_part= ''
+    }
+    $file_build_part = $_version_fields.FileBuildPart
+    If ($file_build_part -eq $null){
+        $file_build_part = ''
+    }
+    $file_private_part = $_version_fields.FilePrivatePart
+    If ($file_private_part -eq $null)
```
Comment (nitzmahone) [inline windows/win_file_version.ps1:67]: This doesn't run - missing a {. Fixing this, works fine.

**Comment by nitzmahone** [comment]
Works fine after fixing the missing { - I'll fix post-merge.

**Pull request merged**